packages/nuxt/src/components/client-fallback-auto-id.ts

```diff
@@ -0,0 +1,52 @@
+import { createUnplugin } from 'unplugin'
+import type { ComponentsOptions } from '@nuxt/schema'
+import MagicString from 'magic-string'
+import { isAbsolute, relative } from 'pathe'
+import { hash } from 'ohash'
+import { isVueTemplate } from './helpers'
+interface LoaderOptions {
+  sourcemap?: boolean
+  transform?: ComponentsOptions['transform'],
+  rootDir: string
+}
+
```

Comment:
Maybe we could extract as a single regexp?

```suggestion
const CLIENT_FALLBACK_RE = /<(ClientFallback|client-fallback)( [^>]*)?>/g
```